src/key_id_header.rs

```diff
@@ -0,0 +1,177 @@
+use bytes::Bytes;
+use itertools::Itertools;
+use protobuf::Message;
+
+use crate::{vector_encryption_metadata::VectorEncryptionMetadata, Error};
+
+// This file is for functions which are working with our key id header value.
+// This value has the following structure:
+// 4 Byte id. This value is a u32 encoded in big endian format.
+// 1 Byte where the first 4 bits are used for which type of edek the id points to (Standalone, Saas Shield, DCP).
+//   The next 4 bits are to denote which type of data follows it (vector metadata, IronCore Edoc, deterministic ciphertext)
+
+const SAAS_SHIELD_EDEK_TYPE_NUM: u8 = 0u8;
+const STANDALONE_EDEK_TYPE_NUM: u8 = 128u8;
+
+type Result<A> = std::result::Result<A, super::Error>;
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct KeyId(pub u32);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum EdekType {
+    SaasShield,
+    Standalone,
+}
+
+impl EdekType {
+    pub(crate) fn to_numeric_value(self) -> u8 {
+        match self {
+            EdekType::SaasShield => SAAS_SHIELD_EDEK_TYPE_NUM,
+            EdekType::Standalone => STANDALONE_EDEK_TYPE_NUM,
+        }
+    }
+
+    pub(crate) fn from_numeric_value(candidate: &u8) -> Result<EdekType> {
+        if candidate == &SAAS_SHIELD_EDEK_TYPE_NUM {
+            Ok(EdekType::SaasShield)
+        } else if candidate == &STANDALONE_EDEK_TYPE_NUM {
+            Ok(EdekType::Standalone)
+        } else {
+            Err(Error::EdekTypeError(
+                "Byte {candidate} isn't a valid edek type.".to_string(),
+            ))
+        }
+    }
+    pub fn create_header(&self, key_id: KeyId) -> Bytes {
+        let mut vec = Vec::with_capacity(6);
+        for b in u32::to_be_bytes(key_id.0) {
+            vec.push(b);
+        }
+        vec.push(self.to_numeric_value());
+        vec.push(0u8);
+        vec.into()
```

Comment:
Thoughts on this? I think they're pretty similar, but I tend towards chaining
```suggestion
        let iter = u32::to_be_bytes(key_id.0)
            .into_iter()
            .chain([self.to_numeric_value(), 0u8]);
        Bytes::from_iter(iter)
```